core/citrus-base/src/test/java/org/citrusframework/functions/core/AdvancedRandomNumberFunctionTest.java

```diff
@@ -0,0 +1,411 @@
+/*
+ * Copyright the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.citrusframework.functions.core;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+import org.citrusframework.context.TestContext;
+import org.citrusframework.exceptions.InvalidFunctionUsageException;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+public class AdvancedRandomNumberFunctionTest {
+
+    private AdvancedRandomNumberFunction function;
+    private TestContext context;
+
+    @BeforeMethod
+    public void setUp() {
+        function = new AdvancedRandomNumberFunction();
+        context = new TestContext();
+    }
+
+    @Test
+    public void testRandomNumberWithNullParameter() {
+        InvalidFunctionUsageException exception = expectThrows(InvalidFunctionUsageException.class,
+            () -> function.execute(null, context));
+        assertEquals(exception.getMessage(),
+            "Function parameters must not be null.");
+    }
+
+    @Test
+    public void testRandomNumberWithDefaultValues() {
+        List<String> params = List.of();
+        String result = function.execute(params, context);
+        assertNotNull(result);
+        assertTrue(result.matches("-?\\d*\\.\\d{2}"));
```

Comment:
just a hint, but `assertJ` could help with things like `isGreaterThanOrEqualTo` and `isLessThanOrEqualTo` to check the bounds. because, as I highlighted in another comment, I think the function is not correctly bound to `Double.MAX_VALUE` and `Double.MIN_VALUE`.

I would also love this to be repeated tests, because the function returns something with a _probability_. you can invoke this using `@Test(invocationCount = 1000)`, I think that should be sufficient. same goes for all tests in this class.

Reply:
It is not bound to these values. I just set some defaults as explained above.

I set an invocation count where appropriate. Internally there have been loops to accomodate for this "random issue". Might have missed some though. Anyway, this class now produces 8800 Tests.

I adjusted some tests and code. So you might want to take a further look.

Reply:
still passes in around 103 ms. all good imo.